src/main/scala/group/scala/karazin/circe/literal/extras/macros.scala

```diff
@@ -3,18 +3,23 @@ package group.scala.karazin.circe.literal.extras
 import java.util.UUID
 
 import cats.implicits.{given, _}
+import cats.implicits._
+import cats.data.{Chain, NonEmptyChain, NonEmptyList, NonEmptyMap, NonEmptySet, NonEmptyVector, OneAnd, Validated}
 import io.circe.parser
 import io.circe.syntax._
 import io.circe.{Json, JsonObject, JsonNumber, Encoder, ACursor, HCursor}
 
 import scala.quoted._
-
 import scala.deriving.Mirror
+import scala.util.{Failure, Success, Try as UTry}
 
 object macros:
 
   object encode:
 
+    final case class EncodeException(private val message: String = "") extends Exception(message)
```

Comment:
Usually case classes do not have private values.
Also according to our logic the message is required.
So let's make it just `EncodeException(message: String)`.